### DISCUSSION TITLE
how to show relative URLs  instead full url in payload

### DISCUSSION THREAD

**Discussion by sanjay-raut** [django-json-api/django-rest-framework-json-api#923] [Q&A]
I am having trouble showing relative links (self, related).
in payload it's always giving me a full URL like `http://localhost:8000/modules/3` , what I want is `/modules/3` instead showing full URL.
how can I achieve it?

**Comment by sliverc** [comment]
In exceptional cases relative urls might make sense but absolute urls are recommend by DRF see [here](https://www.django-rest-framework.org/api-guide/reverse/)

This said you can overwrite `get_url` in your view (maybe create your own project base view) and pass on `request` as `None` so DRF returns relative urls.

This could look like this (just from the top ofof my head without any testing):

```python
def get_url(self, name, view_name, kwargs, request):
  return super().get_url(name, view_name, kwargs, None)
```

Does this work?

**Comment by sanjay-raut** [comment]
@sliverc  no, I  added this in my view but still the same.

we need relative URLs in payload because our APIs are behind API-gateway.

**Comment by sanjay-raut** [comment]
@sliverc  `get_url` is not part of a view, it's a part of `ResourceRelatedField`, so I had to inherit this and override the `get_url` method.
this way I  am able to get relative URLs for related links.

**Comment by sliverc** [comment]
I was thinking you are using relationship view where [get_url](https://github.com/django-json-api/django-rest-framework-json-api/blob/master/rest_framework_json_api/views.py#L247) would need to be overwritten as well.

But of course you also need to do that for `ResourceRelatedField`. Also make sure that your serializers set [serializer_related_field](https://github.com/django-json-api/django-rest-framework-json-api/blob/master/rest_framework_json_api/serializers.py#L229) to the new resource related field.
